code/game/objects/items/weapons/circuitboards/machinery/vending.dm

```diff
@@ -0,0 +1,8 @@
+obj/item/weapon/circuitboard/vending
+	name = "circuit board (custom vendomat)"
+	build_path = /obj/machinery/vending/custom
+	board_type = "machine"
+	origin_tech = list(TECH_DATA = 2)
+	req_components = list(
+		/obj/item/stack/cable_coil = 10
```

Comment:
This should require some more interesting components. At the very least, a matter bin (for storing goods) and a manipulator (for dispensing them)